browser/resources/brave_new_tab_page_refresh/components/settings/settings_modal.tsx

```diff
@@ -0,0 +1,89 @@
+/* Copyright (c) 2025 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+import * as React from 'react'
+import Dialog from '@brave/leo/react/dialog'
+import Navigation from '@brave/leo/react/navigation'
+import NavigationItem from '@brave/leo/react/navigationItem'
+
+import { BackgroundPanel } from './background_panel'
+import { useLocale } from '../context/locale_context'
+
+import { style } from './settings_modal.style'
+
+export type SettingsView =
+  'background'
+
+interface Props {
+  initialView: SettingsView | null
+  isOpen: boolean
+  onClose: () => void
+}
+
+export function SettingsModal(props: Props) {
+  const { getString } = useLocale()
+
+  const [currentView, setCurrentView] =
+    React.useState<SettingsView>(props.initialView || 'background')
+
+  React.useEffect(() => {
+    if (props.isOpen) {
+      setCurrentView(props.initialView || 'background')
+    }
+  }, [props.isOpen, props.initialView])
+
+  function shouldShowView(view: SettingsView) {
+    return true
+  }
+
+  function renderPanel() {
+    if (!shouldShowView(currentView)) {
+      return null
+    }
+    switch (currentView) {
+      case 'background': return <BackgroundPanel />
```

Comment:
This is optional, but I'm generally in favor of having a dictionary over a switch statement as its a lot less verbose (and harder to mess up):

```tsx
// Alternatively you could type it to return a function to make it lazy - it enforces the exhaustive check too
const panels: Record<SettingsView, React.ReactNode> = {
    'background': <BackgroundPanel />
}
```

Reply:
If this was a `Record<SettingsView, React.ReactNode>` defined outside the component then this would work fine (as we wouldn't be recreating the whole tree every render.